mesa: 24.3.4 -> 25.0.1

K900 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

As is tradition, draft before 25.0.1 is out.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

K900 committed:
mesa: 24.3.4 -> 25.0.1

Avoid moving pipe_* to $opencl, because pipe-loader is linked everywhere now

olivia-fl commented on pkgs/development/libraries/mesa/default.nix:
```diff
@@ -227,6 +224,7 @@ in stdenv.mkDerivation {
   ] ++ lib.optionals needNativeCLC [
     (lib.mesonOption "intel-clc" "system")
```
```suggestion
    (lib.mesonOption "intel-clc" "system")
```

I believe intel-clc is no longer necessary, it's handled by precomp-compiler now. The two usage sites are [here][1] and [here][2]. I believe the plan is to remove it in the future. Same thing for `install-intel-clc` and `install-precomp-compiler`.

[1]: https://gitlab.freedesktop.org/mesa/mesa/-/blob/3a8abfa39b73d7fbb68492be864626fdff9c3ebc/meson.build#L323
[2]: https://gitlab.freedesktop.org/mesa/mesa/-/blob/3a8abfa39b73d7fbb68492be864626fdff9c3ebc/meson.build#L323

K900 replied:
Fixed.

mattkang commented:
I think this is going to solve a bug for me that's causing an XWayland segfault on intel. What needs to happen to get this merged?

K900 commented:
Nothing, we're waiting for 25.0.1 as recommended by upstream.

vcunat commented:
> how can I help?

I think more testing never hurts.  It's nice that mesa updates aren't huge rebuilds anymore, so testing is a bit more accessible.

SuperSandro2000 commented on pkgs/development/libraries/mesa/darwin.nix:
```diff
@@ -27,18 +27,6 @@ stdenv.mkDerivation {
 
-  # Darwin build fixes. FIXME: remove in 25.1.
-  patches = [
-    (fetchpatch {
```
Please remove the unused fetchpatch from inputs.

K900 committed:
hey-mail: mesa -> libgbm

K900 committed:
qqmusic: remove unused Mesa input

K900 committed:
mesa-gl-headers: init at 25.0.1

K900 committed:
magpie: use mesa-gl-headers

K900 committed:
muffin: use mesa-gl-headers

K900 committed:
mutter: use mesa-gl-headers

K900 committed:
mutter46: use mesa-gl-headers

K900 committed:
wine: build without osmesa

It's going away soon.

K900 committed:
cogl: use mesa-gl-headers

K900 committed:
enlightenment.efl: use mesa-gl-headers

K900 committed:
arrayfire: mark broken

It is.

K900 committed:
python312Packages.horizon-eda: build without osmesa

K900 committed:
kodi: use mesa-gl-headers

K900 commented:
So, some Shenanigans(tm) later, I've basically achieved my goal of getting Mesa to be 1 rebuild. (it is technically 3, but 1 is primus, which is broken, and the other is xf86-video-vmware, which you should not be using). The changes to the Mesa package itself will also hopefully make sure that nothing picks up this dependency by accident.

K900 committed:
primus: remove spurious Mesa reference

That file doesn't even exist. I'm pretty sure the whole thing is irreparably broken. Oh well.

K900 commented:
OK, some of the shenanigans will probably have to be delayed, as there's _something_ going on with XWayland that I don't fully understand yet. The remaining changes still save a few hundred rebuilds, so they're staying in.

K900 committed:
mesa: output cleanup, part 1

- add sanity check for mesa-gl-headers
- drop osmesa, gallium-nine, xatracker from default build
- move teflon to $drivers
- rename $driversdev to $cross_tools to reflect its real use

K900 committed:
xf86-video-vmware: drop the now useless mesa dependency

SuperSandro2000 commented on pkgs/development/libraries/arrayfire/default.nix:
```diff
@@ -245,5 +245,6 @@ stdenv.mkDerivation rec {
       chessai
       twesterhout
     ];
+    broken = true;
```
That's a regression from master. We should at least add a comment with a short reason.

K900 replied:
It's not? It doesn't build on master either.

K900 replied:
(and hasn't for ages)

jansol commented:
> OK, some of the shenanigans will probably have to be delayed, as there's _something_ going on with XWayland that I don't fully understand yet. The remaining changes still save a few hundred rebuilds, so they're staying in.

Does this mean that it's no longer necessary to build chromium 3 times when touching mesa? That was my main problem with mesa testing before, since that eats massive amounts of memory and takes hours to complete even on a high end system. And QtWebView and different versions of gtk webview all required separate builds...

K900 commented:
This should already not be an issue with the previous batch of libgbm changes, but between this and https://github.com/NixOS/nixpkgs/pull/387292 we should have exactly zero Mesa dependencies left in tree, meaning replacing Mesa will only ever rebuild Mesa.

returntoreality commented:
25.0.1 got released just now

K900 committed:
orca-slicer: get rid of osmesa dependency, clean up other mesa-related things

It's not actually necessary.

K900 commented:
That package was broken long before this change, and osmesa is now gone upstream.
